packages/ng/src/app/select/select.component.ts

```diff
@@ -0,0 +1,184 @@
+import {
```

Comment:
You maybe ;), but others prefer the component. That's why I create both. People like you can use the composition of a directive and popover. And other will use directly the component :)

Reply:
What i dislike most in this compoenent is the fact that it abstractds what is for me a fundamental notion of the LF framework when it comes to inputs, and how is implemented the [SRP](https://en.wikipedia.org/wiki/Single_responsibility_principle) for this problem

You are gonna have to make sure that `<lu-select>` stays this basic stuff for n00bs who dont care, and that anything more complex is gonna need using `<input [luSelect]=...>` otr else you'll end up with a massive lu-select component that does anything and everything

so i would appreciate if it was in a separate module, in a separate route on the demo and not the first thing seen when consulting the demo for `/select`

Reply:
For LF, I have the SRP in mind, and I think that sometimes, you should propose some abstractions to peoples. I'm totally agree with you that, LF should not be the place of <lu-select> <lu-select withOption1> <lu-select withOption2> .... 

So as you can see in the demo, there is the "expert" mode which offer you all the modularity you want and when you want something specific for your app, I will encourage the devs to use, expert mode or to override the basic component. 

Something that i disslike in "directive" approach is the fact it to "angular" and not enought W3C standard. I keep in mind that i will write web component for LF, and a webcomponent is not a directive. I still have some experiments to do for it but i want to be as close as possible of the plaform

Reply:
ok

just sayin' you also could provide the abstraction afterward, not in the first batch. make sure everyone understand the notions before offering a way to bypass it.

> So as you can see in the demo, there is the "expert" mode whi...
 i'd rather this was the first thing seen, and not the `simple` one first

also why did you choose to have this implementation
```html
<lu-select [ngModel]="stuff">
  <lu-select-option>...</>
  <lu-select-option>...</>
</lu-select>
```
and not 
```html
<lu-select [ngModel]="stuff" [options]="options"></>
```
that seems even moar user friendly 360 no scope

Reply:
Because the implementation with <lu-select-option> offers the possiblities to developpers to choose the rendering of an option. We don't need any "callback" function (or interface) to let developpers chose what to show in the list and how.

I do the select with the basic html select in mind, as the select offer in the html is a component, i keep this in mind. In term of css, the html select isn't very customizable

Reply:
whot i mean is that if `lu-select` is the super duper basic stuff that does all for the low price of $19.99, then it shouldn't offer customizable options.
Go all in and offer a component that doesn't offer any customization and direct people to the `expert` version if they want more control, don't offer something that is customizable to a point only

Reply:
i just dont understand the half-measure approach that seems to be at the heart of this compoenent

Reply:
Because it is the same apporach as the html select. In a basic select, you choose what is display in the options. When you manipulate a select, you won't manipulate the `content` the dropdown and the overlay that display the options. You can if you want do the same job than a select by rewriting thoses aspects or you simply use the single component.

In general, for LF, there will be two kind of users, thoses who want to have some "basic all in one" components and thoses who want to manipulate what's under the hood.

When you look at angular material the select is a single component. And with @jeremie-lucca we want to do something like [Polaris of shopify](https://polaris.shopify.com/components/get-started) which offer always two modes, the "expert" and the "all in one".

But be sure that my all in one will be not a "all option of the world in one"

Reply:
> But be sure that my all in one will be not a "all option of the world in one"
i hope its true cuz you're gonna have to pay the bills if it isnt